### PR TITLE
Update documentation to describe IPAM driver

### PR DESCRIPTION
**Pull request by robbrockbank**
Update our documentation and demo instructions for the new Calico libnetwork ipam driver.

### CODE REVIEW

File: docs/getting-started/libnetwork/Demonstration.md [hunk 1]
```diff
@@ -51,29 +51,81 @@ The new command `docker network` can be used to create a logical network.
...
-rather than using the default selections - this is to avoid potential conflicts
-with the default NAT IP assignment used by VirtualBox.  Depending on your
-specific environment, you may need to choose different IP Pool CIDRs.
+#### Networking using Calico IPAM in a non-cloud environment
+
+For Calico IPAM in a non-cloud environment, you need to first create a Calico
+IP Pool with no additional options.  Here we create a pool with CIDR
+192.168.0.0/16.
+
+    calicoctl pool add 192.168.0.0/16
```
Comment (tomdee) [inline docs/getting-started/libnetwork/Demonstration.md]: Won't this have been created already?

File: docs/getting-started/libnetwork/Demonstration.md [hunk 2]
```diff
@@ -129,7 +183,97 @@ fail.
...
+        
+## Accessing Calico's rich policy with libnetwork
+
+Calico networking provides feature rich policy to control access for an 
+endpoint.  However, the Calico libnetwork plugin has some limitations that mean
+it is not possible to access this richer policy using the standard 
+Docker/libnetwork commands..
+
+Notable limitations are:
+-  It is not possible to join a container to more than one network
```
Comment (tomdee) [inline docs/getting-started/libnetwork/Demonstration.md]: I thought this was fine if you're using the libnetwork IPAM?

**Comment by tomdee** [comment]
Looks good other than the section on +## Accessing Calico's rich policy with libnetwork

File: docs/getting-started/libnetwork/Demonstration.md (hunk 1 above)
Comment (robbrockbank) [inline docs/getting-started/libnetwork/Demonstration.md]: Yup - as we discussed, adding a pool simply blats over the existing one.  By instructing to just add the pool is a simplification.  We could say that this pool is created by default so you don't need to do anything, but this is simpler.

File: docs/getting-started/libnetwork/Demonstration.md (hunk 2 above)
Comment (robbrockbank) [inline docs/getting-started/libnetwork/Demonstration.md]: oh crikey - yes I think you're correct - good catch.

**Pull request merged**